src/audio.impl.cpp

```diff
@@ -0,0 +1,269 @@
+#include "audio.impl.hpp"
+
+#include <range/v3/view.hpp>
+#include <range/v3/range.hpp>
+#include <range/v3/algorithm/find_if.hpp>
+
+namespace vencord
+{
+    audio::impl::~impl()
+    {
+        should_exit = true;
+        sender->send(quit{});
+
+        thread.join();
+    }
+
+    audio::impl::impl()
+    {
+        auto [pw_sender, pw_receiver] = pw::channel<pw_recipe>();
+        auto [cr_sender, cr_receiver] = cr::channel<cr_recipe>();
+
+        sender   = std::make_unique<pw_recipe::sender>(std::move(pw_sender));
+        receiver = std::make_unique<cr_recipe::receiver>(std::move(cr_receiver));
+
+        auto thread_start = [this](auto receiver, auto sender)
+        {
+            start(std::move(receiver), std::move(sender));
+        };
+
+        thread = std::thread{thread_start, std::move(pw_receiver), std::move(cr_sender)};
+        receiver->recv_as<ready>();
+    }
+
+    void audio::impl::create_mic()
+    {
+        auto node = core->create<pw::node>({"adapter",
+                                            {{"audio.channels", "2"},
+                                             {"node.name", "Vesktop Screen-Share"},
```

Comment:
a more generic name would be better, since the idea is to make this also be used for Discord Desktop, armcord, etc

Reply:
Name is `Vesktop Screen-Share`, what would you prefer?